### PR TITLE
extract detatched plugin location to a var and allow it to be overridden

### PR DESCRIPTION
**Pull request by jtnord**
Cloudbees products use a custom plugin manager that bundles plugins in `WEB-INF/plugins` so that the product can work even when installed in an airgapped environment.

To prevent multiple restarts on upgrading we also ensure that any plugin in `WEB-INF/detached-plugins` is version aligned with the plugin that we bundle in `plugins`

This causes both bloat and extra overhead.  we do not need to ship the plugins twice, but currently we have to as the location of the `detached-plugin` folder is not configurable.

This PR allows subclasses of `PluginManager` to override the location where Jenkins will look to load the detached plugins from.

### Proposed changelog entries

- Entry 1: Issue, human-readable text
- […]

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)).
  - Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7301"></a>

### CODE REVIEW

**Review by car-roll — APPROVED** [review]

File: core/src/main/java/hudson/PluginManager.java [hunk 1]
```diff
@@ -291,6 +291,11 @@ PluginManager doCreate(@NonNull Class<? extends PluginManager> klass,
         return new LocalPluginManager(jenkins);
     }
 
+    /**
+     * Location of the detacthed plugins in the war
+     */
+    protected detachedPluginLocation = "/WEB-INF/detached-plugins";
```
Comment (jglick) [inline core/src/main/java/hudson/PluginManager.java:297]: Maybe make this a method?

**Review by jglick — APPROVED** [review]
(presuming some downstream PR successfully uses this)

File: core/src/main/java/hudson/PluginManager.java (hunk 1 above)
Comment (car-roll) [reply core/src/main/java/hudson/PluginManager.java:297]:
```suggestion
    protected String detachedPluginLocation = "/WEB-INF/detached-plugins";
```

Comment (car-roll) [reply core/src/main/java/hudson/PluginManager.java:297]: @jglick like this? #7303

**Comment by jtnord** [comment]
closing in favour of #7303

**Pull request closed without merging**